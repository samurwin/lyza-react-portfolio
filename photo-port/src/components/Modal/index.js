import React from 'react';

function Modal({ onClose, photo }) {
    const {name, category, description, index} = photo;
    return (
        <div className='modalBackDrop'>
            <div className='modalContainer'>
                <h3 className='modalTitle'>{name}</h3>
                <img 
                alt='current category'
                src={require(`../../assets/large/${category}/${index}.jpg`)}
                />
                <p>
                    {description}
                </p>
                <button type='button' onClick={onClose}>
                    Close
                </button>
            </div>
        </div>
    );
}

export default Modal;