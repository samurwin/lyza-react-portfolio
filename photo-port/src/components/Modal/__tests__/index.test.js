import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const photo = {
    name: 'Grocery aisle',
    category: 'commercial',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 1
};

const mockToggleModal = jest.fn();

afterEach(cleanup)

describe('Modal', () => {
    it('renders', () => {
        render(<Modal photo={photo} onClose={mockToggleModal} />)
    })

    it('matched snapshot', () => {
        const { asFragment } = render(<Modal photo={photo} onClose={mockToggleModal} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it('click event', () => {
        const { getByText } = render(<Modal photo={photo} onClose={mockToggleModal} />);
        fireEvent.click(getByText('Close'));
        expect(mockToggleModal).toHaveBeenCalled();
    })
})