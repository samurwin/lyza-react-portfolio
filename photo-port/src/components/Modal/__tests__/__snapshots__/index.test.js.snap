// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matched snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackDrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Grocery aisle
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
